lib/DocumentManager.php

```diff
@@ -175,4 +265,34 @@ private function getOptionsResolver($eventName)
 
         return $resolver;
     }
+
+    /**
+     * If the exception is an instanceof `DocumentManagerException` set the
+     * document manager name that threw.
+     *
+     * @param \Exception $e
+     * @param string $context
+     */
+    private function processException(\Exception $exception, $context)
+    {
+        $message = sprintf('Error %s', $context);
+
+        if ($exception instanceof DocumentManagerException) {
+            // if the exception already has a document manager name, then it
+            // was thrown by this method in another document manager instance
+            // so we wrap it again.
+            //
+            // the user-facing exception class will be lost and we are assuming that standard
+            // exceptions (`DocumentNotFoundException`) will not be affected by this
+            // within the scope of normal usage.
+            if ($exception->getDocumentManagerName()) {
+                throw new DocumentManagerException($message, $this->context->getName(), $exception);
```

Comment:
Why does there have to be a new exception created, if the given one already has a DocumentManagerName?